clar2wasm/tests/wasm-generation/control_flow.rs

```diff
@@ -62,3 +64,51 @@ proptest! {
         );
     }
 }
+
+proptest! {
+    #![proptest_config(super::runtime_config())]
+
+    #[test]
+    fn begin(val in PropValue::any()) {
+        crosscheck(
+            &format!(r#"(begin (some {val}) {val})"#),
+            Ok(Some(val.into()))
+        );
+    }
+}
+
+proptest! {
+    #![proptest_config(super::runtime_config())]
+
+    #[test]
+    fn begin_ok(val in PropValue::any()) {
+        crosscheck(
+            &format!(r#"(begin (some {val}) (ok {val}))"#),
+            Ok(Some(Value::Response(ResponseData { committed: true, data: Box::new(val.into()) })))
+        );
+    }
```

Comment:
What is the point of this test? What does it tell us that the previous test doesn’t tell already?

Reply:
So you think intermediary expressions don't even matter? I was testing by wrapping values into `optional` and `response` types. The reason to do this was `(ok 123)` won't work as intermediary expression but `(some (ok 123))` will. I was just checking behavior of wrapped values.

Reply:
I’m not sure I understand your answer.

The intermediary expression is exactly the same in both cases. And you already generate random `ok` values in the Strategy, so the last expression is already tested in the first test.